PWGGA/GammaConvBase/AliConvEventCuts.cxx

```diff
@@ -7808,10 +7897,153 @@ Float_t AliConvEventCuts::GetWeightForMultiplicity(Int_t mult){
+  };
+
+  AliInfo("AliConvEventCuts::GetWeightForMesonNew(): INFO: Starting function\n");
+  if (!indexIsValidAndParticleIsToBeWeighted())
+  {
+    AliWarning(Form("checkSanitizeAndReturnWeight(): WARNING: 1 for particle %d returning 0.\n",
+                      index));
+    return 0.;
+  }
+
+  Double_t mesonPt = 0;
+  Int_t PDGCode = 0;
+  // todo: check why I need to capture everything in order for it work
+  auto getPDGCodeAndMesonPt = [&]()
+  {
+    if (!event || event->IsA() == AliESDEvent::Class())
+    {
+      mesonPt = ((AliMCParticle *)mcEvent->GetTrack(index))->Pt();
+      PDGCode = ((AliMCParticle *)mcEvent->GetTrack(index))->PdgCode();
+    }
+    else if (event->IsA() == AliAODEvent::Class())
+    {
+      if (!fAODMCTrackArray)
+        fAODMCTrackArray = dynamic_cast<TClonesArray *>(event->FindListObject(AliAODMCParticle::StdBranchName()));
+      if (fAODMCTrackArray)
+      {
+        AliAODMCParticle *aodMCParticle = static_cast<AliAODMCParticle *>(fAODMCTrackArray->At(index));
+        mesonPt = aodMCParticle->Pt();
+        PDGCode = aodMCParticle->GetPdgCode();
+      }
+      else
+      {
+        return 0;
+      }
+    }
+    return 1;
+  };
+
+  if (!getPDGCodeAndMesonPt())
+  {
+    AliWarning(Form("checkSanitizeAndReturnWeight(): WARNING: 2 for meson %d. returning 0.\n",
+                      PDGCode));
 
+    return 0.;
+  }
+
+  auto checkSanitizeAndReturnWeight = [&](Double_t theWeight)
+  {
+    if ((theWeight < 0) || !isfinite(theWeight))
+    {
+      theWeight = 0.;
+      AliWarning(Form("checkSanitizeAndReturnWeight(): WARNING: Weight for meson %d is negative or not finite: %f.\n"
+      " It was set to 0 - effectively rejecting the particle.\n"
+                      "This points to a severe problem - investigate!\n",
+                      PDGCode, theWeight));
+    }
+    return theWeight;
+  };
+
+  // catch cases with invalid PDGCode
+  auto const &lConstIt = fMapPtWeightsAccessObjects.find(PDGCode);
+  if (lConstIt == fMapPtWeightsAccessObjects.cend())
+  {
+    AliWarning(Form("GetWeightForMesonNew(): WARNING: 3: PDGCode %d not found in fMapPtWeightsAccessObjects. Returning 1.\n", PDGCode));
+    return 1.;
+  }
+
+  Double_t lNomData = lConstIt->second.fData->Eval(mesonPt);
+  Double_t lDenomMC = lConstIt->second.hMC->Interpolate(mesonPt);
+  auto calcWeight = [&checkSanitizeAndReturnWeight, &lNomData, &lDenomMC]()
+  {
+    Double_t lWeight = lDenomMC
+                           ? (lNomData > 0.)
+                                 ? lNomData / lDenomMC
+                                 : 0 // to signal problem
+                           : -1.;    // to signal problem      
+    // will reset to 1 and throw a warning if weight is not >=0 and finite
+    return checkSanitizeAndReturnWeight(lWeight);
+  };
+
+  bool lCaseEtaPi0 = (PDGCode == 111) || (PDGCode == 221);
+  double lResult = lCaseEtaPi0
+             ? calcWeight()
+             : checkSanitizeAndReturnWeight(lDenomMC);
```

Comment:
So if the PDGCode is neither Pi0 nor eta, why do we need to run `checkSanitizeAndReturnWeight(lDenomMC)`? why not directly return 0 or something?

Reply:
good catch. This was a remnant from before the early exit in case particle not found in map. Back then, I used it to issue a warning and reset to 1 now indeed it was completely superflous. I fixed it, thanks :)